.github/workflows/release.yml

```diff
@@ -0,0 +1,83 @@
+name: Bump Version, Create Tag, and Release
+
+on: 
+  workflow_dispatch:
+    inputs:
+      versionBump:
+        description: 'Version bump type'
+        required: true
+        default: 'minor'
+        type: choice
+        options:
+          - patch
+          - minor
+          - major
+
+env:
+      MARKET_TOKEN: ${{ secrets.MARKETPLACE_ACCESS_TOKEN }}
+
+permissions:
+  contents: write
+
+jobs:
+   deploy:
+    runs-on: ubuntu-latest
+    steps:
+    
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Install Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: 16.x
+
+    - name: Install dependencies
+      run: npm ci
+
+    - name: Configure Git
+      run: |
+        git config user.name "GitHub Actions"
+        git config user.email "github-actions@env.com"
+
+    - name: Upgrade package version
+      run: npm version ${{ github.event.inputs.versionBump }} -m "Upgrade to %s"
+
+    - name: Push changes
+      run: |
+        git push
+        git push --tags
```

Comment:
not sure why you push tags here and not at "Create Git Tag" step @meniRoy

Reply:
I am sorry, I forgot to remove this code bc71c2138462fa068cd220752b066e1b2ebb907b.
Since you use NPM version it checks for us if the tag exists, and it fails, and it also creating a tag for the new version